pkg/cache/v3/delta.go

```diff
@@ -22,50 +22,27 @@ import (
 	"github.com/envoyproxy/go-control-plane/pkg/server/stream/v3"
 )
 
-// Respond to a delta watch with the provided snapshot value. If the response is nil, there has been no state change.
-func respondDelta(ctx context.Context, request *DeltaRequest, value chan DeltaResponse, state stream.StreamState, snapshot Snapshot, log log.Logger) (*RawDeltaResponse, error) {
-	resp, err := createDeltaResponse(ctx, request, state, snapshot)
-	if err != nil {
-		if log != nil {
-			log.Errorf("Error creating delta response: %v", err)
-		}
-		return nil, nil
-	}
+type resourceContainer struct {
+	resourceMap map[string]types.Resource
 
-	// Only send a response if there were changes
-	// We want to respond immediately for the first wildcard request in a stream, even if the response is empty
-	// otherwise, envoy won't complete initialization
-	if len(resp.Resources) > 0 || len(resp.RemovedResources) > 0 || (state.IsWildcard() && state.IsFirst()) {
-		if log != nil {
-			log.Debugf("node: %s, sending delta response with resources: %v removed resources %v wildcard: %t",
-				request.GetNode().GetId(), resp.Resources, resp.RemovedResources, state.IsWildcard())
-		}
-		select {
-		case value <- resp:
-			return resp, nil
-		case <-ctx.Done():
-			return resp, context.Canceled
-		}
-	}
-	return nil, nil
-}
+	versionMap map[string]string
 
-// nolint:unparam // result 1 (error) is always nil (unparam)
-func createDeltaResponse(ctx context.Context, req *DeltaRequest, state stream.StreamState, snapshot Snapshot) (*RawDeltaResponse, error) {
-	resources := snapshot.GetResources((req.TypeUrl))
+	systemVersion string
+}
 
+func createDeltaResponse(ctx context.Context, req *DeltaRequest, state stream.StreamState, resources *resourceContainer, log log.Logger) *RawDeltaResponse {
 	// variables to build our response with
 	nextVersionMap := make(map[string]string)
-	filtered := make([]types.Resource, 0, len(resources))
+	filtered := make([]types.Resource, 0, len(resources.resourceMap))
 	toRemove := make([]string, 0)
 
 	// If we are handling a wildcard request, we want to respond with all resources
 	switch {
 	case state.IsWildcard():
-		for name, r := range resources {
+		for name, r := range resources.resourceMap {
 			// Since we've already precomputed the version hashes of the new snapshot,
 			// we can just set it here to be used for comparison later
-			version := snapshot.GetVersionMap()[req.TypeUrl][name]
```

Comment:
I'm guessing the resource container is going to be created on a per type basis now? Since we don't key off the type anymore from the snapshot